osrf/mbzirc
# Add Docker files for cloudsim

depends on #126 

Adds 3 more Dockerfiles for a total of 4 docker images:
* `mbzirc_models`: base image with mbzirc models downloaded from fuel
* `mbzirc_sim`: existing docker image, modified to build on top of `mbzirc_models`
* `cloudsim_sim`: builds on top of `mbzirc_sim` with `ENTRYPOINT` that calls a `run_sim.bash` script for launching sim
* `cloudsim_bridge`: builds on top of `mbzirc_sim` with `ENTRYPOINT` that calls a `run_bridge.bash` script for launching a bridge per robot

To test:

I have already built and pushed a [mbzirc_models](https://hub.docker.com/layers/224783651/osrf/mbzirc/mbzirc_models_2022-05-31_2349/images/sha256-538a4f5b29656c99083cbbc330cdc531fc9650ed497a3ec13b30e871b64d4609?context=repo)  docker image to dockerhub for testing

To build the other docker images:

```sh
cd mbzirc
bash docker/build.bash mbzirc_sim
bash docker/build.bash cloudsim_sim
bash docker/build.bash cloudsim_bridge
```

Below shows the intended usage. Note that a `robot` arg is added to `spawn_config.launch.py` to let the bridge container launch only the bridges associated with a particular robot

```sh
# launch cloudsim_sim
bash run.bash cloudsim_sim "world:=coast config_file:=/home/developer/mbzirc_ws/src/mbzirc/mbzirc_ign/config/coast/hexrotor.yaml sim_mode:=sim"

# launch cloudsim_bridge for hexrotor_1 robot
bash run.bash cloudsim_bridge "world:=coast config_file:=/home/developer/mbzirc_ws/src/mbzirc/mbzirc_ign/config/coast/hexrotor.yaml robot:=hexrotor_1 sim_mode:=bridge"
```
